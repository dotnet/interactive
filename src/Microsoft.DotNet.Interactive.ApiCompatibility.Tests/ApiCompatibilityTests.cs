// Copyright (c) .NET Foundation and contributors. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using System.Diagnostics;
using Assent;
using Microsoft.DotNet.Interactive.CSharp;
using Microsoft.DotNet.Interactive.CSharpProject;
using Microsoft.DotNet.Interactive.Documents;
using Microsoft.DotNet.Interactive.Formatting;
using Microsoft.DotNet.Interactive.FSharp;
using Microsoft.DotNet.Interactive.HttpRequest;
using Microsoft.DotNet.Interactive.Journey;
using Microsoft.DotNet.Interactive.Jupyter;
using Microsoft.DotNet.Interactive.Kql;
using Microsoft.DotNet.Interactive.Mermaid;
using Microsoft.DotNet.Interactive.PowerShell;
using Microsoft.DotNet.Interactive.SqlServer;
using Microsoft.DotNet.Interactive.Tests.Utility;
using Xunit;

namespace Microsoft.DotNet.Interactive.ApiCompatibility.Tests;

public class ApiCompatibilityTests
{
    private readonly Configuration _configuration;

    public ApiCompatibilityTests()
    {
        _configuration = new Configuration()
                         .SetInteractive(Debugger.IsAttached)
                         .UsingExtension("txt");
    }

    [FactSkipLinux("Testing api contract changes, not needed on Linux too")]
    public void Interactive_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<Kernel>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux("Testing api contract changes, not needed on Linux too")]
    public void Formatting_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<FormatContext>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux("Testing api contract changes, not needed on Linux too")]
    public void Document_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<InteractiveDocument>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux("Testing api contract changes, not needed on Linux too")]
    public void PackageManagement_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<PackageRestoreContext>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux("Testing api contract changes, not needed on Linux too")]
    public void Journey_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<Lesson>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux("Testing api contract changes, not needed on Linux too")]
    public void csharp_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<CSharpKernel>();
        this.Assent(contract, _configuration);
    }

    [Fact(Skip = "this api is in early design stage.")]
    public void csharpProject_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<CSharpProjectKernel>();
        this.Assent(contract, _configuration);
    }

    [Fact(Skip = "need to use signature files")]
    public void fsharp_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<FSharpKernel>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux("Testing api contract changes, not needed on Linux too")]
    public void powershell_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<PowerShellKernel>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux("Testing api contract changes, not needed on Linux too")]
    public void mssql_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<MsSqlKernelConnector>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux("Testing api contract changes, not needed on Linux too")]
    public void kql_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<KqlKernelConnector>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux("Testing api contract changes, not needed on Linux too")]
    public void mermaid_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<MermaidKernel>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux("Testing api contract changes, not needed on Linux too")]
    public void jupyter_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<ConnectionInformation>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux("Testing api contract changes, not needed on Linux too")]
    public void httpRequest_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<HttpRequestKernel>();
        this.Assent(contract, _configuration);
    }
}