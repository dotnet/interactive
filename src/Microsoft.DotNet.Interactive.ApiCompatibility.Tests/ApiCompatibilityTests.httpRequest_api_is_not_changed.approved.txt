Microsoft.DotNet.Interactive.HttpRequest
  public class HttpRequestKernel : Microsoft.DotNet.Interactive.Kernel, Microsoft.DotNet.Interactive.IKernelCommandHandler<Microsoft.DotNet.Interactive.Commands.RequestDiagnostics>, Microsoft.DotNet.Interactive.IKernelCommandHandler<Microsoft.DotNet.Interactive.Commands.RequestKernelInfo>, Microsoft.DotNet.Interactive.IKernelCommandHandler<Microsoft.DotNet.Interactive.Commands.SendValue>, Microsoft.DotNet.Interactive.IKernelCommandHandler<Microsoft.DotNet.Interactive.Commands.SubmitCode>, System.IDisposable
    .ctor(System.String name = null, System.Net.Http.HttpClient client = null)
    public System.Uri BaseAddress { get; set;}
    public System.Collections.Generic.IEnumerable<System.ValueTuple<System.String,System.Int32,System.Int32>> GetAllRequests(System.String requests)
    public System.Threading.Tasks.Task HandleAsync(Microsoft.DotNet.Interactive.Commands.SendValue command, Microsoft.DotNet.Interactive.KernelInvocationContext context)
    public System.Threading.Tasks.Task HandleAsync(Microsoft.DotNet.Interactive.Commands.SubmitCode command, Microsoft.DotNet.Interactive.KernelInvocationContext context)
    public System.Threading.Tasks.Task HandleAsync(Microsoft.DotNet.Interactive.Commands.RequestDiagnostics command, Microsoft.DotNet.Interactive.KernelInvocationContext context)
    public System.Void SetValue(System.String valueName, System.String value)
   class HttpRequest
    .ctor(System.String verb, System.String address, System.String body, System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<System.String,System.String>> headers)
    public System.String Address { get;}
    public System.String Body { get;}
    public System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<System.String,System.String>> Headers { get;}
    public System.String Verb { get;}
  public class HttpRequestKernelExtension, Microsoft.DotNet.Interactive.IKernelExtension
    .ctor()
    public System.Threading.Tasks.Task OnLoadAsync(Microsoft.DotNet.Interactive.Kernel kernel)
